packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -225,6 +226,18 @@ mixin SchedulerBinding on BindingBase, ServicesBinding {
   AppLifecycleState get lifecycleState => _lifecycleState;
   AppLifecycleState _lifecycleState;
 
+  /// Initializes the [lifecycleState] with the data from the window.
+  ///
+  /// An optional [testWindow] may be provided to supply test values.
+  ///
+  /// Once the lifecycleState is initialized, it cannot be initialized again.
+  @protected
+  void initLifecycleState() {
```

Comment:
Chatted offline, but I would recommend a name for this along the lines of `readLifecycleStateFromNativeWindow()` or `readLifecycleStateFromHostPlatform()`. This explicitly reveals what it means to "init".